test/Npgsql.Tests/PoolTests.cs

```diff
@@ -405,15 +469,14 @@ public void ExercisePool(int maxPoolSize, int numTasks, int seconds, bool async)
 
         volatile int StopFlag;
 
-        void AssertPoolState(ConnectorPool? pool, int idle, int busy, int waiting=0)
+        void AssertPoolState(ConnectorPool? pool, int open, int idle)
```

Comment:
Did you remove the ability to assert on waiting because of timing issue/race conditions? Because it seems like a useful assert to me.

Reply:
There's no waiting state, so it was already not there for a long while. I can allow internal access to the queue length in a follow up PR?

Reply:
Fixed in this PR